### PR TITLE
docs: Fix Go distributed tracing example

### PR DESCRIPTION
**Pull request by carsonip**
## Motivation/summary

Fix Go distributed tracing examples docs.

## Checklist

- [ ] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/main/CHANGELOG.asciidoc)
- [ ] Update [package changelog.yml](https://github.com/elastic/apm-server/blob/main/apmpackage/apm/changelog.yml) (only if changes to `apmpackage` have been made)
- [ ] Documentation has been updated

For functional changes, consider:
- Is it observable through the addition of either **logging** or **metrics**?
- Is its use being published in **telemetry** to enable product improvement?
- Have system tests been added to avoid regression?

### CODE REVIEW

**Comment by carsonip** [comment]
This is potentially be backported very far. How far do we want to go?

File: docs/tab-widgets/distributed-trace-receive.asciidoc [hunk 1]
```diff
@@ -21,7 +21,7 @@ traceContext.State, _ = apmhttp.ParseTracestateHeader(r.Header["Tracestate"]...)
 opts := apm.TransactionOptions{
 	TraceContext: traceContext, <3>
 }
-transaction := apm.DefaultTracer.StartTransactionOptions("GET /", "request", opts) <4>
+transaction := apm.DefaultTracer().StartTransactionOptions("GET /", "request", opts) <4>
```
Comment (axw) [inline docs/tab-widgets/distributed-trace-receive.asciidoc:24]: @carsonip we should also update the godoc.org links above to pkg.go.dev, and reference the /v2 import paths

**Review by axw — APPROVED** [review]

**Comment by carsonip** [comment]
I believe this can be backported to 7.17 and 8.x without conflict.

**Review by bmorelli25 — APPROVED** [review]
Thank you.

**Pull request merged**